amgut/test/test_auth_handlers.py

```diff
@@ -0,0 +1,133 @@
+                                 {'kit_id': unregistered_kits[0],
+                                  'password': 'test',
+                                  'email': 'e@mail',
+                                  'email2': 'e@mail',
+                                  'participantname': 'joe',
+                                  'address': 'street',
+                                  'city': 'metropolis',
+                                  'state': 'XX',
+                                  'zip': '12345',
+                                  'country': 'Afghanistan'})
+            port = self.get_http_port()
+            self.assertEqual(response.code, 200)
+            self.assertEqual(
+                response.effective_url,
+                'http://localhost:%d/?next=%s' %
+                (port, url_escape('/authed/portal/')))
+
+            # Check that registration changed the DB. A proxy to test that no
+            # ValueError is raised.
+            self.assertNotEqual(ag_data.get_user_for_kit(unregistered_kits[0]),
+                                'something')
+
+
+class TestAuthLoginHandler(TestHandlerBase):
+    def test_get(self):
+        response = self.get('/auth/login/')
+        self.assertEqual(response.code, 200)
+        # make sure redirect happened properly
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url, 'http://localhost:%d/' % port)
+
+        # check if url is wrong. SJ: Is this in general a good idea?
+        response = self.get('/authh/login/')
+        self.assertEqual(response.code, 404)
+
+    def test_post_correct_pass(self):
+        # test a sucessful login, for a handout kit
+        self.mock_login('tst_ACJUJ')
+        response = self.get('/authed/portal/')
+        self.assertEqual(response.code, 200)
+        self.assertNotIn(text_locale['handlers']['REGISTER_KIT'],
+                         response.body)
+
+        # test a sucessful login, for a non-handout kit
+        self.mock_login('tst_aAdKt')
+        response = self.get('/authed/portal/')
+        self.assertEqual(response.code, 200)
+        self.assertIn(text_locale['portal.html']['VERIFICATION_HEADER_2'],
+                      response.body)
+
+    def test_post_wrong_pass(self):
+        # check that invalid kit is reported for non existent skid
+        response = self.post('/auth/login/', {'skid': 'wrong',
+                                              'passwd': 'test'})
+        self.assertIn(text_locale['handlers']['INVALID_KITID'], response.body)
+        self.assertEqual(response.code, 200)
+
+        # check that invalid password is reported for wrong password
+        response = self.post('/auth/login/', {'skid': 'tst_ACJUJ',
+                                              'passwd': 'wrong'})
+        self.assertIn(text_locale['handlers']['INVALID_KITID'], response.body)
+        self.assertEqual(response.code, 200)
+
+    def test_set_current_user(self):
```

Comment:
I think you are right. It is basically an artifact from the template I was using to write my code, i.e. it is qiita's auth_handler unit test.